texas-hearing-institute/App.tsx

```diff
@@ -1,21 +1,103 @@
 import { StatusBar } from 'expo-status-bar';
 import { StyleSheet, Text, View } from 'react-native';
-import tw from 'tailwind-react-native-classnames';
+import ToggleGridButtons from './components/ToggleGridButtonsComponent/ToggleGridButtons';
+import { useState, useEffect } from 'react';
+import { storageKey } from './pages/Constants';
+// test with the Stack Navigator for React-Native
+import { NavigationContainer } from '@react-navigation/native';
+import { createNativeStackNavigator } from '@react-navigation/native-stack';
+
+// Persistent storage. This is just a test and not expected to be permanent
+import AsyncStorage from '@react-native-async-storage/async-storage';
+import Onboarding1 from './pages/Onboarding1';
+import Onboarding2 from './pages/Onboarding2';
+import Onboarding3 from './pages/Onboarding3';
+import Onboarding4 from './pages/Onboarding4';
+
+
 
 export default function App() {
-  return (
-    <View style={styles.container}>
-      <Text style={tw`text-3xl font-bold underline`}>Open up App.tsx to start working on your app!</Text>
-      <StatusBar style="auto" />
-    </View>
-  );
-}
+    const testItems = ['b', 'm', 'w', 'f', 'v', 'th', 'TH', 't', 'd', 'n', 's', 'z', 'l', 'sh', 'zh', 'ch', 'dg', 'y', 'r', 'k', 'g', 'ng', 'h'];
+    const [itemsSelected, setItemsSelected] = useState(Array(testItems.length).fill(false));
+
+    // Store selection
+    const storeItemSelection = async (storageKey: string, items: string[], itemsSelected: boolean[]) => {
+        let itemSelectionMap = new Map<string, boolean>();
+        for (let i = 0; i < items.length; i++) {
+            itemSelectionMap.set(items[i], itemsSelected[i]);
+        }
+        try {
+            await AsyncStorage.setItem(storageKey, JSON.stringify(Array.from(itemSelectionMap.entries())));
+        } catch (e) {
+            console.log(e);
+        }
+    }
+
+    // Retrieve selection
+    const retrieveItemSelections = async (storageKey: string, items: string[]): Promise<boolean[]> => {
+        try {
+            const itemSelectionJson = await AsyncStorage.getItem(storageKey);
+            let itemSelectionMap = new Map<string, boolean>();
+
+            if (itemSelectionJson !== null) {
+                const itemSelectionArray: [string, boolean][] = JSON.parse(itemSelectionJson);
+                itemSelectionMap = new Map(itemSelectionArray);
+            }
 
+            const itemsSelected: boolean[] = items.map(item => itemSelectionMap.get(item) ?? false);
+            return itemsSelected;
+
+        } catch (e) {
+            console.log(e);
+            return items.map(() => false);
+        }
+    }
+
+    // Load stored selection statuses when the component mounts
+    useEffect(() => {
```

Comment:
useEffect should be used for things like side effects; I think it is better design to load in the components in useState itself; please try making this change throughout all the files (I have sent you a screenshot on Slack with an example of how to use async storage within a useState)